tests/e2e/steps/ViewerContext.ts

```diff
@@ -0,0 +1,98 @@
+import { Given, When, Then, DataTable } from '@cucumber/cucumber'
+import { expect } from '@playwright/test'
+import config from '../config'
+
+import { Ocis } from '../pageObjects/Ocis'
+import { Viewer } from '../pageObjects/Viewer'
+import { uploadFile } from '../utils/helpers'
+import util from 'util'
+
+const ocis = new Ocis()
+const viewer = new Viewer()
+
+Given(
+  'the following 3D models have been uploaded:',
+  async function (filesForUpload: DataTable): Promise<void> {
+    for (const file of filesForUpload.hashes()) {
+      await uploadFile(file.filename)
+      // assert if file is listed in the file list after upload
+      const locator = await global.page
+        .locator(util.format(ocis.elements.resourceNameSelector, file.filename))
+        .toString()
+      expect(locator).toContain(file.filename)
+    }
+  }
+)
+
+Given(
+  'the user has logged in with username {string} and password {string}',
+  async function (user: string, password: string): Promise<void> {
+    await global.page.goto(config.baseUrlOcis)
+    await ocis.login({ username: user, password: password })
+  }
+  // to do: assert that the user has successfully logged in
```

Comment:
why it's to do?

Reply:
can't adding `  await page.locator('#web-content').waitFor()` satisfy

Reply:
Thanks for the suggestion!
todo because I didn't had time to look into it yet. It's a reminder for me to implement some suggestions Phil was giving me when chatting F2F about this :)

Reply:
proper assertion has been added to this `Given` step